src/topology/algebra/ring.lean

```diff
@@ -7,32 +7,47 @@ Theory of topological rings.
 -/
 import topology.algebra.group
 import ring_theory.ideal.basic
+import algebra.ring.prod
 
 open classical set filter topological_space
 open_locale classical
 
 section topological_ring
 universes u v w
-variables (α : Type u) [topological_space α]
+variables (α : Type u)
 
 /-- A topological semiring is a semiring where addition and multiplication are continuous. -/
-class topological_semiring [semiring α]
+class topological_semiring [topological_space α] [semiring α]
   extends has_continuous_add α, has_continuous_mul α : Prop
 
-variables [ring α]
-
 /-- A topological ring is a ring where the ring operations are continuous. -/
-class topological_ring extends has_continuous_add α, has_continuous_mul α : Prop :=
+class topological_ring [topological_space α] [ring α]
+  extends has_continuous_add α, has_continuous_mul α : Prop :=
 (continuous_neg : continuous (λa:α, -a))
 
-variables [t : topological_ring α]
+variables {α}
+
+/-- The product topology on the cartesian product of two topological semirings
+  makes the product into a topological semiring. -/
+instance prod_semiring {β : Type u}
+  [semiring α] [topological_space α] [topological_semiring α]
+  [semiring β] [topological_space β] [topological_semiring β] : topological_semiring (α × β) :=
+{}
+
+variables [ring α] [topological_space α] [t : topological_ring α]
 @[priority 100] -- see Note [lower instance priority]
 instance topological_ring.to_topological_semiring : topological_semiring α := {..t}
 
 @[priority 100] -- see Note [lower instance priority]
 instance topological_ring.to_topological_add_group : topological_add_group α := {..t}
 
-variables {α} [topological_ring α]
+variables [topological_ring α]
+
+/-- The product topology on the cartesian product of two topological rings
+  makes the product into a topological ring. -/
+instance prod_ring {β : Type u}
```

Comment:
```suggestion
instance prod_ring {β : Type v}
```
or
```suggestion
instance prod_ring {β : Type*}
```
otherwise this forces them to be from the same universe

Reply:
Oh! That explains a lot of issues I was having. Thank you.